cluster/aws/configure-vm.sh

```diff
@@ -0,0 +1,676 @@
+}
+
+function download-release() {
+  # In case of failure checking integrity of release, retry.
+  until try-download-release; do
+    sleep 15
+    echo "Couldn't download release. Retrying..."
+  done
+
+  echo "Running release install script"
+}
+
+function fix-apt-sources() {
+  sed -i -e "\|^deb.*http://http.debian.net/debian| s/^/#/" /etc/apt/sources.list
+  sed -i -e "\|^deb.*http://ftp.debian.org/debian| s/^/#/" /etc/apt/sources.list.d/backports.list
+}
+
+function node-docker-opts() {
+  if [[ -n "${EXTRA_DOCKER_OPTS-}" ]]; then
+    DOCKER_OPTS="${DOCKER_OPTS:-} ${EXTRA_DOCKER_OPTS}"
+  fi
+
+  # Decide whether to enable a docker registry mirror. This is taken from
+  # the "kube-env" metadata value.
+  if [[ -n "${DOCKER_REGISTRY_MIRROR_URL:-}" ]]; then
+    echo "Enable docker registry mirror at: ${DOCKER_REGISTRY_MIRROR_URL}"
+    DOCKER_OPTS="${DOCKER_OPTS:-} --registry-mirror=${DOCKER_REGISTRY_MIRROR_URL}"
+  fi
+}
+
+function run-user-script() {
+  if curl-metadata k8s-user-startup-script > "${INSTALL_DIR}/k8s-user-script.sh"; then
+    user_script=$(cat "${INSTALL_DIR}/k8s-user-script.sh")
+  fi
+  if [[ ! -z ${user_script:-} ]]; then
+    chmod u+x "${INSTALL_DIR}/k8s-user-script.sh"
+    echo "== running user startup script =="
+    "${INSTALL_DIR}/k8s-user-script.sh"
+  fi
+}
+
+function ensure-apparmor-service() {
+  local hostfqdn=$(hostname -f)
+  local hostname=$(hostname)
+  echo "127.0.1.1 $hostfqdn $hostname" >> /etc/hosts
+
+  # Start AppArmor service before we have scripts to configure it properly
+  if ! sudo systemctl is-active --quiet apparmor; then
+    echo "Starting Apparmor service"
+    sudo systemctl start apparmor
+  fi
+}
+
+function setup-flannel-cni-conf() {
+  mkdir -p /etc/cni/net.d
+  cat > /etc/cni/net.d/10-flannel.conflist <<EOF
+{
+  "name": "cbr0",
+  "plugins": [
+    {
+      "type": "flannel",
+      "delegate": {
+        "hairpinMode": true,
+        "isDefaultGateway": true
+      }
+    },
+    {
+      "type": "portmap",
+      "capabilities": {
+        "portMappings": true
+      }
+    }
+  ]
+}
+EOF
+}
+
+function enable-root-ssh() {
+  sudo sed -i 's/.*sleep 10\" //' /root/.ssh/authorized_keys
+}
+
+function ensure-containerd() {
+  echo "Installing containerd .."
+  pushd /tmp
+  sudo wget https://storage.googleapis.com/cri-containerd-release/cri-containerd-1.3.3.linux-amd64.tar.gz
+  tar --no-overwrite-dir -C / -xzf cri-containerd-1.3.3.linux-amd64.tar.gz
+  systemctl daemon-reload
+  systemctl start containerd
+  popd
+}
+
+function ensure-docker() {
+  echo "Installing docker .."
+  DEBIAN_FRONTEND=noninteractive sudo apt-get purge -y docker docker-engine docker.io
+  DEBIAN_FRONTEND=noninteractive sudo apt-get install -y apt-transport-https ca-certificates curl gnupg2 software-properties-common
+  curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
+  sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian stretch stable"
```

Comment:
Why debian?

Reply:
i was running into some interactive stops, googled and found this to be a working solution. if there's a better way, i can change it.

Reply:
i removed this and found it working without issues now after moving from debian to ubuntu.